cmake/Modules/GetDDSConf.cmake

```diff
@@ -0,0 +1,22 @@
+#
+# Copyright(c) 2021 ADLINK Technology Limited and others
+#
+# This program and the accompanying materials are made available under the
+# terms of the Eclipse Public License v. 2.0 which is available at
+# http://www.eclipse.org/legal/epl-2.0, or the Eclipse Distribution License
+# v. 1.0 which is available at
+# http://www.eclipse.org/org/documents/edl-v10.php.
+#
+# SPDX-License-Identifier: EPL-2.0 OR BSD-3-Clause
+#
+
+function(GET_DDSCONF _executable _depends)
+  if (CMAKE_CROSSCOMPILING)
+    find_program(_EXE ddsconf REQUIRED NO_CMAKE_FIND_ROOT_PATH)
+    set(${_executable} ${_EXE} PARENT_SCOPE)
+    set(${_depends} "" PARENT_SCOPE)
+  else()
+    set(${_executable} CycloneDDS::ddsconf PARENT_SCOPE)
+    set(${_depends} CycloneDDS::ddsconf PARENT_SCOPE)
+  endif()
+endfunction()
```

Comment:
I see why you'd be tempted to do it like this, but I think that if you apply what we do for `idlc` in the place where the `ddsconf` target is created. That'd work? Perhaps (and this might be a nicer solution for `idlc` too), you can create a custom target that is either an alias of `CycloneDDS::ddsconf` or a custom target. That can probably also be used in the `DEPENDS` for the custom commands too(?)